wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceService.java

```diff
@@ -184,10 +188,14 @@ public WorkspaceDto getByKey(
                 })
           )
           @PathParam("key")
-          String key)
+          String key,
+      @ApiParam("Servers with which visibility to show")
+          @DefaultValue("external")
+          @QueryParam("visibility")
+          Visibility visibility)
       throws NotFoundException, ServerException, ForbiddenException, BadRequestException {
     validateKey(key);
-    return asDtoWithLinksAndToken(workspaceManager.getWorkspace(key));
+    return filterVisible(asDtoWithLinksAndToken(workspaceManager.getWorkspace(key)), visibility);
```

Comment:
When I see `visibility=internal` I expect to get servers which have internal visibility. But in fact, a response will contain all servers (including internal servers). Maybe it would be more logic to use here another parameter for including internal servers

Reply:
After discussion with @sleshchenko and @gazarenkov we decided to rename query parameter to `includeInternalServers` with no value or boolean. ServerConfig attribute should be renamed to `internal`:"<true|false>". If it is missing it is supposed that server is external.